dashboard/app/controllers/lessons_controller.rb

```diff
@@ -232,4 +232,19 @@ def cachable_request?(request)
       ProgrammingExpression.find_by!(programming_environment: environment, key: e['key'])
     end
   end
+
+  private def get_unit(params)
+    # /s/.../lessons/... URL
+    return Unit.get_from_cache(params[:script_id]) if params[:script_id]
+    # /courses/.../unit/.../lessons/...
+    course_name = params[:course_course_name]
+    if course_name
+      course = UnitGroup.get_from_cache(course_name)
+      unit_position = params[:unit_position]
+      raise ActiveRecord::RecordNotFound unless course && unit_position
+      unit_group_unit = UnitGroupUnit.find_by(course_id: course.id, position: unit_position)
+      return Unit.get_from_cache(unit_group_unit.script_id) if unit_group_unit
+    end
+    raise ActiveRecord::RecordNotFound
```

Comment:
after pulling down your branch, I'm hitting an error on this line when I go to the following url:
* http://localhost-studio.code.org:3000/courses/csd-2024/units/3/lessons/2/levels/1

The following URLs are all working:
* http://localhost-studio.code.org:3000/s/csd3-2024/
* http://localhost-studio.code.org:3000/courses/csd-2024/
* http://localhost-studio.code.org:3000/s/csd3-2024/lessons/2/levels/1

is there something I'm missing?

Reply:
apologies, the error is happening not here, but at this location: https://github.com/code-dot-org/code-dot-org/blob/4f7280f69d3e144b2f0b8a4d296a4166f5aa6f52/dashboard/app/controllers/script_levels_controller.rb#L372